src/Uno.UI/UI/Xaml/Media/Animation/Storyboard.cs

```diff
@@ -114,16 +114,15 @@ public void Begin()
 
 		private void Play()
 		{
-			_runningChildren = 0;
-			if (Children != null && Children.Count > 0)
+			_runningChildren = Children?.Count ?? 0;
+			if (_runningChildren > 0)
```

Comment:
Is the change about `Children` being changed in a reentrant while Play is being executed? It cannot (should not) be a concurrency, but an ordering issue, I'd think.

Reply:
No, the change in the `if` line is irrelevant refactoring to avoid the duplicate null check. The main fix is the initialization of `_runningChildren` to the whole count instead of 0.

What was previously happening is:

- `_runningChildren = 0`
- Increment, now `_runningChildren` is 1.
- Start `RepositionThemeAnimation`.
- `RepositionThemeAnimation` will raise that it has changed very fast (because it's not implemented and just calls `OnCompleted`
- Then, `ChildCompleted` is called, which decrements `_runningChildren`.
- Now `_runningChildren` is zero, and the storyboard thinks it has completed.
- But then, other *transition animations* will continue to run and increment `_runningChildren`.
- Since the storyboard raised its complete event, then it will start to execute the setters.
- At this point, I think the setters and transitions are executing concurrently and racing with each other.